packages/react/src/Trans.tsx

```diff
@@ -10,17 +10,16 @@ export type TransRenderProps = {
 }
 
 export type TransRenderType =
-  | string
-  | React.ElementType<TransRenderProps>
-  | React.ReactElement
```

Comment:
We need to refactor this function a bit. First of all, if we remove support for `string`, `React.ElementType<TransRenderProps>` and `React.ReactElement` we also need to remove the if-branches which handles these cases and related tests:

https://github.com/lingui/js-lingui/blob/7b0ff4cba08d9b2d6d13e5c2632f8a84b20904b7/packages/react/src/Trans.tsx#L66-L74

Also, this is going to be a breaking change which we need to document.

---

Finally, I don't think we can support `React.FunctionComponent<TransRenderProps>` here. Since we're calling the function directly, I believe fucntion component won't work as ithey might contain hooks and possible side effects.

Reply:
Alright, so i'll take a look more in depth with your notes

Reply:
I did some changes Tomáš, to keep the compatibility with string and added a new `if` to check if the render prop is a component or a class, so we use the current method `React.createElement`, and if the user is just using `<Trans render={(props) => <Element {...props />`, we use the `render({})` alternative

I've added some testing to check that the is still rendering the content correctly